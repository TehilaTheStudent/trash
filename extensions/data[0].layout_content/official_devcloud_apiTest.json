[
  {
    "componentName": "d-form",
    "id": 1665824682303,
    "category": "form",
    "props": {
      "layout": "vertical",
      "ref": "formRef"
    },
    "children": [
      {
        "componentName": "d-form-item",
        "id": 1668218195895,
        "category": "form",
        "props": {
          "label": "Select Task",
          "field": "jobId",
          "help-tips": "",
          "rules": [
            {
              "required": true,
              "message": "This field cannot be left blank.",
              "trigger": [
                "blur",
                "change"
              ]
            },
            {
              "regex": "",
              "message": "请选择需要调用的任务内容不符合规范",
              "trigger": [
                "blur",
                "change"
              ]
            }
          ],
          "visible": {
            "type": "fixed",
            "conditions": [],
            "value": true
          },
          "disabled": false
        },
        "children": [
          {
            "componentName": "link-select",
            "id": 1668218195894,
            "props": {
              "selectType": "normal",
              "singleOptions": {
                "type": "api",
                "fixedOptions": [
                  {
                    "label": "选项",
                    "value": "1",
                    "defaultChecked": false,
                    "visible": false
                  }
                ],
                "apiOptions": {
                  "url": "/v2/pipeline/getPipelineSubTaskResult",
                  "method": "GET",
                  "params": {
                    "projectId": "${project_id}",
                    "taskType": "apiTest",
                    "recordNum": "20"
                  },
                  "label": "jobName",
                  "value": "id",
                  "body": {},
                  "responseUrl": "result.jobResultList",
                  "linkedFields": [],
                  "remote": true,
                  "remoteName": "jobName",
                  "remoteQueryField": "params"
                }
              },
              "isUpdateChildAll": true,
              "remote": "",
              "allow-clear": false,
              "filter": "",
              "linkText": "Create One",
              "linkHref": "/cloudtestportal/project/${project_id}/testsuite?type=1&creat=true",
              "herfTips": "",
              "disabled": {
                "type": "fixed",
                "conditions": [],
                "value": false
              }
            },
            "category": "form"
          }
        ]
      },
      {
        "componentName": "d-form-item",
        "id": 1666948012161,
        "category": "form",
        "props": {
          "label": "Task parameters",
          "field": "key_1666948012160",
          "help-tips": "",
          "rules": null,
          "visible": {
            "type": "fixed",
            "conditions": [],
            "value": true
          },
          "disabled": false
        },
        "children": [
          {
            "componentName": "task-parameter",
            "id": 1666948012159,
            "props": {
              "params": {
                "type": "api",
                "apiOptions": {
                  "url": "/v2/pipeline/getPipelineSubTaskParamWithContext",
                  "method": "POST",
                  "params": {
                    "taskType": "apiTest",
                    "subTaskID": "${jobId}",
                    "projectId": "${project_id}",
                    "taskName": "Task_10"
                  },
                  "label": "",
                  "value": "",
                  "body": "",
                  "responseUrl": "result.parameters",
                  "linkedFields": [
                    "jobId"
                  ],
                  "remote": false,
                  "remoteName": "",
                  "remoteQueryField": "params"
                }
              },
              "isCustomVar": false,
              "disabled": {
                "type": "fixed",
                "conditions": [],
                "value": false
              }
            },
            "category": "form"
          }
        ]
      },
      {
        "componentName": "d-form-item",
        "id": 1669087180306,
        "category": "form",
        "props": {
          "label": "metrics",
          "field": "SYSTEM_METRICS_UNIQUE_KEY",
          "help-tips": "",
          "visible": {
            "type": "fixed",
            "conditions": [],
            "value": true
          },
          "disabled": false
        },
        "children": [
          {
            "componentName": "d-input",
            "id": 1669087180305,
            "props": {
              "inputType": "metrics",
              "placeholder": "请输入",
              "isTrim": true,
              "autocomplete": "off",
              "defaultValue": "/opt/devcloud/workspace",
              "disabled": {
                "type": "fixed",
                "value": true,
                "conditions": []
              },
              "front_data_prop": [
                {
                  "group_name": "APITest-Pass-Rate-Gate",
                  "properties": [
                    {
                      "key": "pass_ratio",
                      "value": "0.5",
                      "desc": "PassRatio"
                    }
                  ]
                }
              ]
            },
            "category": "form"
          }
        ]
      }
    ]
  }
]