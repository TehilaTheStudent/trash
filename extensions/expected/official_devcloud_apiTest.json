[
  {
    "key": "environmentParaModel",
    "value": "1"
  },
  {
    "key": "globalEnvironmentParaId",
    "value": "9002ca6e68f54e8da77e58e4a4ee3858"
  },
  {
    "key": "jobId",
    "value": "vg0n00010i1o22pi"
  },
  {
    "key": "_OFFICIAL_DEVCLOUD_JOB_NAME_",
    "value": "Create API Test Automation Test Suite"
  }
]